frontend/src/metabase/lib/query/field.js

```diff
@@ -21,26 +22,38 @@ export function getFieldTargetId(field: Field): ?FieldId {
     console.warn("Unknown field type: ", field);
 }
 
-export function isRegularField(field: Field): boolean {
+export function isRegularField(field: FieldReference): boolean {
     return typeof field === "number";
 }
 
-export function isLocalField(field: Field): boolean {
+export function isLocalField(field: FieldReference): boolean {
     return Array.isArray(field) && mbqlEq(field[0], "field-id");
 }
 
-export function isForeignKeyField(field: Field): boolean {
+export function isForeignKeyField(field: FieldReference): boolean {
     return Array.isArray(field) && mbqlEq(field[0], "fk->");
 }
 
-export function isDatetimeField(field: Field): boolean {
+export function isDatetimeField(field: FieldReference): boolean {
     return Array.isArray(field) && mbqlEq(field[0], "datetime-field");
 }
 
-export function isExpressionField(field: Field): boolean {
+export function isExpressionField(field: FieldReference): boolean {
     return Array.isArray(field) && field.length === 2 && mbqlEq(field[0], "expression");
 }
 
-export function isAggregateField(field: Field): boolean {
+export function isAggregateField(field: FieldReference): boolean {
     return Array.isArray(field) && mbqlEq(field[0], "aggregation");
 }
+
+// https://github.com/metabase/metabase/issues/3417
+export function getFieldValues(field: ?Field): any[] {
```

Comment:
Would be nice to have a comment about the behavior, not only a link to GitHub issue